### PR TITLE
Fix inconsistent dependency caps

### LINKED ISSUE #2430: Fix unnecessary dependency caps
**Is your feature request related to a problem? Please describe.**
@kevin-bates raised some good points in [#2429 (comment)](https://github.com/elyra-ai/elyra/pull/2429#pullrequestreview-869586789) that we should address with a few of the dependency caps.

The below items are directly from the relevant review comment:
- `jupyter_core` (5.0) cap should be removed - we're close enough to that repo to be aware of any breaking major releases.

- `minio` (8.0 is a non-existent release) - this cap was updated from 7 when a contribution required version 7, implying there was no reason for it to be capped at 7 in the first place. The contributor added 8, I'm sure, due to precedence. Btw, the same PR that moved the `minio` cap to 8.0 also updated the kfp base to 1.7.0, so the comment regarding `click` and `kfp==1.6.3` is moot anyway.

- `pyyaml` (6.0) - This cap looks like it might be legit as 6.0 does (now) exist. However, if we use `yaml.safe_load()` - which appears to be the case, then `safe_load()` calls `load()` with the `SafeLoader` loader - so all is good.

- `requests` (3.0 is a non-existent release) - However, their docs indicate that major releases will contain breaking changes. As a result, I think they view releases differently in that new features that are backward compatible will not be a major release.

**Describe the solution you'd like**
Fix the relevant dependencies accordingly.

### PR DESCRIPTION
**Pull request by kiersten-stokes**
Closes #2430 

### What changes were proposed in this pull request?
Removes the cap for:

- `jupyter_core` (former cap on v5 was unnecessary as Elyra works close enough with this repo to be aware of breaking changes)

- `minio` (former cap on v8 didn't exist)

- `pyyaml` (former cap on v6 should not introduce any breaking changes; tested and confirmed)

- `requests` (former cap on v3 didn't exist; future releases, however, may include breaking changes)

### How was this pull request tested?
Tests are passing. Also manually tested to ensure that `pyyaml` v6 has the expected behavior.

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -77,23 +77,23 @@
...
         'nbconvert>=5.6.1',
         'nbdime~=3.1',
         'nbformat>=5.1.2',
         'networkx>=2.5.1',
         'papermill>=2.1.3',
         'python-lsp-server[all]>=1.1.0',
-        'pyyaml>=5.3.1,<6.0',
-        'requests>=2.25.1,<3.0',
+        'pyyaml>=5.3.1',
+        'requests>=2.25.1',  # Note: future major releases of `requests` are expected to include breaking changes
```
Comment (ptitzler) [inline setup.py:96]: Is there a specific source for this statement that makes it necessary to state this explicitly?

Comment (kiersten-stokes) [reply setup.py:96]: That was my source as well, but I was misreading it -- I kept substituting the phrase 'backwards incompatibility' with breaking change 🙄 . I'll go ahead and remove!

**Review by ptitzler — APPROVED** [review]
LGTM

**Review by kevin-bates — APPROVED** [review]
LGTM - thanks.

**Pull request merged**